src/ci/github-actions/jobs.yml

```diff
@@ -406,6 +405,14 @@ auto:
       RUST_CONFIGURE_ARGS: --build=x86_64-pc-windows-msvc --enable-lld
     <<: *job-windows
 
+  - image: x86_64-msvc-ext3
```

Comment:
Could you please add a comment here explaining why are the jobs split, and what is this specific job doing (checking tools and uploading toolstate)? Our jobs are critically undocumented, so if we're making changes, we might as well leave it in a better state than before.

Reply:
Done 👍
I'm retrying bors because I removed an env variable from the first job, too.